airflow/providers/amazon/aws/hooks/redshift_sql.py

```diff
@@ -104,7 +104,9 @@ def get_iam_token(self, conn: Connection) -> tuple[str, str, int]:
         port = conn.port or 5439
         # Pull the custer-identifier from the beginning of the Redshift URL
         # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
-        cluster_identifier = conn.extra_dejson.get("cluster_identifier", conn.host.split(".")[0])
+        cluster_identifier = conn.extra_dejson.get("cluster_identifier", None)
+        if conn.host and not cluster_identifier:
+            cluster_identifier = conn.host.split(".")[0]
```

Comment:
maybe we need a proper exception if both are not present

Reply:
Though I wonder why cluster identifier wasn't just put in the host.

Being able to do `redshift://my cluster/db?iam=true` would be nice (maybe that boat has already sailed?)

Reply:
Added an exception for the case when both are not passed here: https://github.com/apache/airflow/pull/31567/commits/4e643c39459bedabd94106632d205b2b2278145c